File: data_structures/linked_list/linkedlist.js
class Node {
  constructor(value) {
    this.value = value
    this.next = null
  }
}

class LinkedList {
  constructor(head = null) {
    this.head = head
    this.length = 0
  }

  // Will Append New Node at the end of the list
  append(node) {
    let current = this.head
    let newNode = new Node(node)

    // If head is empty new node is the head
    if (!this.head) {
      this.head = newNode
      this.length++
      return this.head
    }

    while (current.next) {
      current = current.next
    }
    current.next = newNode
    this.length++
    return newNode
  }

  // Will Prepend New Node at the beginning of the list
  prepend(node) {
    let newNode = new Node(node)

    newNode.next = this.head
    this.head = newNode
    this.length++
    return this.head
  }

  // WILL FIND A NODE AT A POSITION
  find(position) {
    let current = this.head
    let counter = 1
    while (counter <= position && current.next) {
      if (counter === position) {
        return current
      }
      current = current.next
      counter++
    }
    return 'Not Found'
  }

  // WILL INSERT A NEW NODE AT THE SPECIFIED POSITION
  insertAt(node, position) {
    const newNode = new Node(node)
    let current = this.head
    let counter = 1
    if (position <= 1) {
      this.head = newNode
      return newNode
    }

    while (counter <= position && current.next) {
      if (counter === position - 1) {
        newNode.next = current.next
        current.next = newNode
        return newNode
      }
      current = current.next
      counter++
    }
  }

  // WILL DELETE THE NODE WITH A GIVEN VALUE
  delete(value) {
    let current = this.head
    let previous = null

    while (current.value !== value && current.next) {
      previous = current
      current = current.next
    }

    if (current.value === value) {
      if (previous) {
        previous.next = current.next
        this.length--
        return previous
      }

      this.length--
      this.head = current.next
    }
  }
}

File: data_structures/linked_list/linkedList.py
#!/usr/bin/env python
class Node():
    def __init__(self, value):
        self.value = value
        self.next = None

class LinkedList():
    def __init__(self, head=None):
        self.head = head

    def append(self, value):
        current = self.head
        if self.head:
            while current.next:
                current = current.next
            current.next = value
        else:
            self.head = value

    def prepend(self, value):
        current = self.head
        value.next = current
        self.head = value

    def list_len(self):
        current = self.head
        count = 0
        while current:
            count += 1
            current = current.next
        return count

    def find_position(self, position):
        current = self.head
        counter = 1
        if position < counter:
            return None
        elif position == counter:
            return self.head
        else:
            while current:
                if position == counter:
                    return current
                else:
                    counter += 1
                    current = current.next

    def insert(self, new_element, position):
        current = self.head
        counter = 1
        while counter < position:
            if counter == position-1:
                new_element.next = current.next
                current.next = new_element
            counter += 1
            current = current.next

    def delete(self, value):
        current = self.head
        previous = None
        while current.value != value and current.next:
            previous = current
            current = current.next
        if current.value == value:
            if previous:
                previous.next = current.next
            else:
                self.head = current.next